src/main/java/no/fintlabs/adapter/ResourceSubscriber.java

```diff
@@ -9,38 +9,36 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
+import java.util.concurrent.Flow;
 import java.util.stream.Collectors;
 
+import static java.util.concurrent.Flow.Subscriber;
+
 @Slf4j
-public abstract class ResourceHandler {
+public abstract class ResourceSubscriber<T extends FintLinks, P extends ResourcePublisher<T, ResourceRepository<T>>>
+        implements Subscriber<List<T>> {
 
     private final WebClient webClient;
     protected final AdapterProperties adapterProperties;
-    protected boolean started;
 
 
-    protected ResourceHandler(WebClient webClient, AdapterProperties adapterProperties) {
+    protected ResourceSubscriber(WebClient webClient, AdapterProperties adapterProperties, P publisher) {
         this.webClient = webClient;
         this.adapterProperties = adapterProperties;
-    }
 
-    public void start() {
-        log.info("Started sync service for {}.", getCapability().getEntityUri());
-        started = true;
-        onFullSync();
+        publisher.subscribe(this);
     }
 
-    public void stop() {
-        log.info("Stopped sync service for {}.", getCapability().getEntityUri());
-        started = false;
+    public void onSync(List<T> resources) {
+        log.info("Syncing {} items to endpoint {}", resources.size(), getCapability().getEntityUri());
+        getPages(resources, 500).
```

Comment:
The pageSize should probably be input from the implementation? So different subscribers can have different page sizes.

Reply:
This should probably be moved to the `AdapterCapability` model. Then we can do `getCapability().getPageSize()`